src/ApiException.php

```diff
@@ -138,9 +216,36 @@ public static function createFromRestApiResponse(
             $previous
         );
     }
+    
+    /**
+     * Checks if decoded metadata includes errorInfo message.
+     * @param array $decodedMetadata
+     * @return array $errorInfo {
+     *     @type boolean $containsErrorInfo
+     *     @type int|null $keyOfErrorInfo
+     * }
+     */
+    public static function containsErrorInfo(array $decodedMetadata)
+    {
+        $errorInfo = array();
+        $errorInfo['containsErrorInfo'] = false;
+        if (empty($decodedMetadata)) {
+            return $errorInfo;
+        }
+        foreach ($decodedMetadata as $key=>$value) {
+            if (array_key_exists('@type', $value) and ($value['@type'] === 'google.rpc.errorinfo-bin' or $value['@type'] === 'type.googleapis.com/google.rpc.ErrorInfo') or (array_key_exists('reason', $value) && array_key_exists('domain', $value) && array_key_exists('metadata', $value))) {
```

Comment:
So this is checking "Any containing an ErrorInfo OR array with ErrorInfo fields" ?

For readability, you might consider breaking these into two variables like so:
```php
$isErrorInfoAny = array_key_exists('@type', $value) &&
    ($value['@type'] === 'google.rpc.errorinfo-bin' ||
     $value['@type'] === 'type.googleapis.com/google.rpc.ErrorInfo');
$isErrorInfoArray = array_key_exists('reason', $value) &&
    array_key_exists('domain', $value) &&
    array_key_exists('metadata', $value);

if ($isErrorInfoAny || $isErrorInfoArray) {
```